mesonbuild/modules/python.py

```diff
@@ -297,6 +299,52 @@ def __init__(self, interpreter: 'Interpreter') -> None:
             'find_installation': self.find_installation,
         })
 
+    def _get_install_scripts(self) -> T.List[mesonlib.ExecutableSerialisation]:
+        ret = []
+        installdata = self.interpreter.backend.create_install_data()
+        py_files: T.Dict[str, T.List[str]] = collections.defaultdict(list)
+        optlevel = self.interpreter.environment.coredata.get_option(mesonlib.OptionKey('bytecompile', module='python'))
+        if optlevel == -1:
+            return ret
+
+        def should_append(f, isdir: bool = False):
+            return f.startswith(('{py_platlib}', '{py_purelib}')) and (f.endswith('.py') or isdir)
+
+        for t in installdata.targets:
+            if should_append(t.out_name):
+                py_files[t.subproject].append(os.path.join(installdata.prefix, t.outdir, os.path.basename(t.fname)))
+        for d in installdata.data:
+            if should_append(d.install_path_name):
+                py_files[d.subproject].append(os.path.join(installdata.prefix, d.install_path))
+        for d in installdata.install_subdirs:
+            if should_append(d.install_path_name, True):
+                py_files[d.subproject].append(os.path.join(installdata.prefix, d.install_path, ''))
+        pycompile = os.path.join(self.interpreter.environment.get_scratch_dir(), 'pycompile.py')
+        if os.path.exists(pycompile):
+            os.remove(pycompile)
+        for i in self.installations.values():
+            if isinstance(i, PythonExternalProgram) and i.run_bytecompile[i.info['version']]:
+                import importlib.resources
+
+                i = T.cast(PythonExternalProgram, i)
+                manifest = f'python-{i.info["version"]}-installed.json'
+                manifest_json: T.List[str] = []
+                with open(pycompile, 'wb') as f:
+                    f.write(importlib.resources.read_binary('mesonbuild.scripts', 'pycompile.py'))
+                for s in py_files:
+                    for f in py_files[s]:
+                        if f.startswith((os.path.join(installdata.prefix, i.platlib), os.path.join(installdata.prefix, i.purelib))):
+                            manifest_json.append(f)
+                with open(os.path.join(self.interpreter.environment.get_scratch_dir(), manifest), 'w', encoding='utf-8') as f:
+                    json.dump(manifest_json, f)
+                cmd = i.command + [pycompile, manifest, str(optlevel)]
+                script = self.interpreter.backend.get_executable_serialisation(cmd, verbose=True)
+                ret.append(script)
+        return ret
+
+    def postconf_hook(self, b: Build) -> None:
+        b.install_scripts.extend(self._get_install_scripts())
```

Comment:
Doing it at build time invites two questions:
- where to store the compiled files
- how to determine what install_subdir() files to compile

Also since we have to apply this to arbitrary targets with `install_dir: py.get_install_dir()` we'd have to append new build rules after the fact, and appending install rules just feels slightly less awkward.

Also keeping in mind #10779 I guess making these into build rules would result in locking us out of providing such a "--backend=none" since bytecompile would then require producing ninja rules.

> Likewise I added a while ago `meson install --strip` because there is no reason to reconfigure the whole project to change the strip option if it's used only at install time.

I believe it was wrong to add that option, but I'm not going to make a huge stink about it.

However, I think that adding dedicated `meson install` options for *module* options is going too far.

Reply:
Hm, I guess if it does not take significant CPU time like compiling C/C++ then it is acceptable. After all, no optimization is basically just stripping comments if I understood correctly.